addons/calendar/models/calendar_event.py

```diff
@@ -435,10 +435,11 @@ def create(self, vals_list):
         # Prevent sending update notification when _inverse_dates is called
         self = self.with_context(is_calendar_event_new=True)
 
-        vals_list = [  # Else bug with quick_create when we are filter on an other user
-            dict(vals, user_id=self.env.user.id) if not 'user_id' in vals else vals
-            for vals in vals_list
-        ]
+        for i, vals in enumerate(vals_list):
+            if 'user_id' not in vals:  # Else bug with quick_create when we are filter on an other user
+                vals_list[i] = dict(vals, user_id=self.env.user.id)
```

Comment:
this could be written without needing the enumerate as far as I can tell:

```suggestion
            if 'user_id' not in vals: 
                vals['user_id'] = self.env.user.id
```
or using setdefault (that set the value if it's not set yet):

```suggestion
            vals.setdefault('user_id', self.env.user.id)
```